File: backend/src/main/java/com/example/bookacourt/dtos/FilterCriteria.java
package com.example.bookacourt.dtos;

import com.example.bookacourt.enums.BookingStatus;
import com.example.bookacourt.enums.CourtEnvironment;
import com.example.bookacourt.enums.CourtType;
import com.example.bookacourt.enums.GreekCity;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.Instant;

@Data
@AllArgsConstructor
public class FilterCriteria {
    private String searchTerm;
    private BookingStatus status;
    private GreekCity city;
    private CourtType courtType;
    private CourtEnvironment environment;
    private Double minPrice;
    private Double maxPrice;
    private Long courtId;
    private Instant dateFrom;
    private Instant dateTo;
    private boolean admin;
}

File: frontend/src/dtos/FilterCriteria.ts
import type { CourtEnvironment, CourtType } from "../enums/CourtEnums";
import type { GreekCity } from "../enums/CourtEnums";
import type { BookingDTO } from "./Booking";

export interface FilterCriteria {
  searchTerm?: string;
  status?: BookingDTO["status"];
  city?: GreekCity;
  courtType?: CourtType;
  environment?: CourtEnvironment;
  minPrice?: number;
  maxPrice?: number;
  courtId?: number;
  dateFrom?: string; // ISO date format (YYYY-MM-DD) for ZonedDateTime parsing
  dateTo?: string; // ISO date format (YYYY-MM-DD) for ZonedDateTime parsing
  page?: number;
  size?: number;
  admin: boolean;
}
